distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1618,6 +1658,172 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
                 .resolve(CSW_SOURCE_ID + "-" + metacardId + ".ser")), is(false));
     }
 
+    @Test
+    public void testTwoUsersSameProductRetrySuccess() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL));
+        CometDClient cometDClient2 = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL));
+
+        setupCswServerForSuccess(cometDClient1, filename);
+        setupCswServerForSuccess(cometDClient2, filename);
+
+        // where 3 is the number of expectations and 9 is the number of activities
+        checkExpectations(cometDClient1, 3, 9);
+        checkExpectations(cometDClient2, 3, 9);
```

Comment:
`3` and `9` are magic numbers ... create a `final int clearly_describing_constant_name = 3` at the top of this method (class scope only if necessary; that is, you can reuse between methods).